modules/apps/portal/portal-upgrade-impl/src/main/java/com/liferay/portal/upgrade/internal/log/UpgradeLogAppender.java

```diff
@@ -105,13 +103,15 @@ public boolean isStopped() {
 	public void setHandler(ErrorHandler handler) {
 	}
 
-	public void setUpgradeReport(UpgradeReport upgradeReport) {
-		_upgradeReport = upgradeReport;
-	}
-
 	@Override
 	public void start() {
-		_started = true;
+		if (PropsValues.UPGRADE_REPORT_ENABLED) {
```

Comment:
@achaparro @SamZiemer 
Generally we do not use `PropsValues` directly in a component, use `Props` service by `@Reference` instead.
In your case, as you have a portal properties, and the entrance is in a portal-impl class `DBUpgrader`
, I would prefer to add the switch logic in `DBUpgrader`, use it to decide to create `ServiceDependencyManager` or not.